dao/mongodb.rya/src/main/java/mvm/rya/mongodb/MongoDBRyaDAO.java

```diff
@@ -179,85 +145,85 @@ public void destroy() throws RyaDAOException {
+            }
+        } catch (final IOException e) {
+            log.error("Unable to add: " + statement.toString());
+            throw new RyaDAOException(e);
+        }
+    }
+
+    @Override
+    public void add(final Iterator<RyaStatement> statement) throws RyaDAOException {
+        final List<DBObject> dbInserts = new ArrayList<DBObject>();
+        while (statement.hasNext()){
+            final RyaStatement ryaStatement = statement.next();
+            final DBObject insert = storageStrategy.serialize(ryaStatement);
+            dbInserts.add(insert);
+
             try {
-                for (RyaSecondaryIndexer index : secondaryIndexers) {
+                for (final RyaSecondaryIndexer index : secondaryIndexers) {
                     index.storeStatement(ryaStatement);
                 }
-            } catch (IOException e) {
-                throw new RyaDAOException(e);
+            } catch (final IOException e) {
+                log.error("Failed to add: " + ryaStatement.toString() + " to the indexer");
             }
-            
-		}
-		coll.insert(dbInserts, new InsertOptions().continueOnError(true));
-	}
-
-	public void delete(RyaStatement statement, MongoDBRdfConfiguration conf)
-			throws RyaDAOException {
-		DBObject obj = storageStrategy.getQuery(statement);
-		coll.remove(obj);
-	}
-
-	public void dropGraph(MongoDBRdfConfiguration conf, RyaURI... graphs)
-			throws RyaDAOException {
-		
-	}
-
-	public void delete(Iterator<RyaStatement> statements,
-			MongoDBRdfConfiguration conf) throws RyaDAOException {
-		while (statements.hasNext()){
-			RyaStatement ryaStatement = statements.next();
-			coll.remove(storageStrategy.getQuery(ryaStatement));
-		}
-		
-	}
-
-	public String getVersion() throws RyaDAOException {
-		return "1.0";
-	}
-
-	public RyaQueryEngine<MongoDBRdfConfiguration> getQueryEngine() {
-		return queryEngine;
-	}
-
-	public RyaNamespaceManager<MongoDBRdfConfiguration> getNamespaceManager() {
-		return nameSpaceManager;
-	}
-
-	public void purge(RdfCloudTripleStoreConfiguration configuration) {
-		// TODO Auto-generated method stub
-		
-	}
-
-	public void dropAndDestroy() throws RyaDAOException {
-		db.dropDatabase(); // this is dangerous!
-	}
 
+        }
+        coll.insert(dbInserts, new InsertOptions().continueOnError(true));
+    }
+
+    @Override
+    public void delete(final RyaStatement statement, final MongoDBRdfConfiguration conf)
+            throws RyaDAOException {
+        final DBObject obj = storageStrategy.getQuery(statement);
+        coll.remove(obj);
+    }
+
+    @Override
+    public void dropGraph(final MongoDBRdfConfiguration conf, final RyaURI... graphs)
+            throws RyaDAOException {
 
+    }
+
+    @Override
+    public void delete(final Iterator<RyaStatement> statements,
+            final MongoDBRdfConfiguration conf) throws RyaDAOException {
+        while (statements.hasNext()){
+            final RyaStatement ryaStatement = statements.next();
```

Comment:
that wasn't originally in the dao and isn't part of the task.  Since there isn't any current way to get to the secondary indexes without going through the DB directly using literals that may require more changes